src/jdk.jlink/share/classes/jdk/tools/jlink/plugin/ResourcePoolEntry.java

```diff
@@ -68,7 +68,21 @@ public enum Type {
         MAN_PAGE,
         NATIVE_CMD,
         NATIVE_LIB,
-        TOP
+        TOP;
+
+        public static Type fromOrdinal(int value) {
```

Comment:
Can be dropped if `JRTArchive.ResourceFileEntry` keeps a map from ordinal to Type instances.